### ISSUE TITLE
Build failure on macOS

### ISSUE THREAD

**Issue by gibfahn** [nodejs/node-report#32]
- macOS Version: `Darwin Kernel Version 16.1.0` (Sierra)
 - clang Version: `Apple LLVM version 8.0.0 (clang-800.0.42.1)`

### To reproduce:

```bash
git clone https://github.com/nodejs/nodereport.git && cd nodereport
npm i
```

### Error

```bash
➜  nodereport git:(master) npm i
npm WARN deprecated node-uuid@1.4.7: use uuid module instead

> nodereport@1.0.7 install /Users/gib/tmp/nodereport
> node-gyp rebuild

gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory '/Library/Developer/CommandLineTools' is a command line tools instance

gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
  CXX(target) Release/obj.target/nodereport/src/node_report.o
../src/node_report.cc:282:21: error: no member named 'prev' in namespace 'std'
    if (it != *std::prev(comp_versions.end())) {
               ~~~~~^
1 error generated.
make: *** [Release/obj.target/nodereport/src/node_report.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/Users/gib/.nvm/versions/node/v6.7.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Darwin 16.1.0
gyp ERR! command "/Users/gib/.nvm/versions/node/v6.7.0/bin/node" "/Users/gib/.nvm/versions/node/v6.7.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/gib/tmp/nodereport
gyp ERR! node -v v6.7.0
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok 

npm ERR! Darwin 16.1.0
npm ERR! argv "/Users/gib/.nvm/versions/node/v6.7.0/bin/node" "/Users/gib/.nvm/versions/node/v6.7.0/bin/npm" "i"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.3
npm ERR! code ELIFECYCLE
npm ERR! nodereport@1.0.7 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nodereport@1.0.7 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the nodereport package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs nodereport
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls nodereport
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/gib/tmp/nodereport/npm-debug.log
```

cc/ @richardlau

**Comment by richardlau** [comment]
@gibfahn This seems to be fixed by applying nodejs/nodereport#31.

**Comment by gibfahn** [comment]
@richardlau Yep, you're right

**Comment by rnchamberlain** [comment]
Fix landed in github, not published in npm yet.

**Comment by hhellyer** [comment]
@rnchamberlain It looks like #31  fix isn't compatible with Node 4/5. The xcode settings don't work with Node 4/5.
I've got a fix which removes the deprecation warning when Node is > v5 and also removes std::prev, I'll push it once I've had a chance to verify it.

**Comment by richardlau** [comment]
This was fixed by #56 and released in [node-report v2.0.0](https://github.com/nodejs/node-report/releases/tag/v2.0.0).
